ingenialink/ethercat/network.py

```diff
@@ -516,23 +519,27 @@ def _notify_status(self, slave_id: int, status: NET_DEV_EVT) -> None:
         for callback in self.__observers_net_state[slave_id]:
             callback(status)
 
-    def _recover_from_power_cycle(self, status: NET_DEV_EVT) -> None:
-        """Transition the slaves to Pre-Op state after a power cycle.
+    def _recover_from_disconnection(self) -> bool:
+        """Recover the CoE communication after a disconnection.
+        All the connected slaves need to transitioned to the PreOp state.
 
-        Args:
-            status: The network status.
-
-        Raises:
-            ILStateError: If not all slaves reach PreOp state.
```

Comment:
From the function name I would expect the exception to be thrown

Reply:
We don't need to raise an exception, we need to retry if the communication cannot be recovered.